handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -363,7 +363,11 @@ public List getRequestedServerNames() {
                 // setMode may impact the overhead.
                 calculateMaxWrapOverhead();
             } catch (Throwable cause) {
-                SSL.freeSSL(ssl);
+                // Call shutdown so we are sure we correctly release all native memory and also guard against the
+                // case when shutdown() will be called by the finalizer again. If we would call SSL.free(...) directly
+                // the finalizer may end up calling it again as we would miss to update the DESTROYED_UPDATER.
+                shutdown();
```

Comment:
@normanmaurer I thought the reference counted open ssl engine didn't use a finalizer?

Reply:
That’s true but the OpenSslEngine does which extends ReferenceCountedOpenSslEngine